lib/ControlledVocab/ControlledVocab.js

```diff
@@ -357,7 +366,9 @@ class ControlledVocab extends React.Component {
       />
     );
 
-    const rows = this.filteredRows(this.props.resources.values.records || []);
+    const rows = this.parseRows(
+      this.filteredRows(this.props.resources.values.records || [])
```

Comment:
Don't you think that `resources` could be `null` at some point? Maybe it makes sense to put here `_.get(this.props.resources, 'values.records', [])`

Reply:
agreed

Reply:
just noticed that we have https://github.com/folio-org/stripes-smart-components/pull/516/files#diff-6a43094b2f04c9ea138bb80cc1d5352dR357

with it `_.get` doesn't make sense here

Reply:
That's OK, but in case it's not loaded, `values` might exists, and records could be undefined, right?

Reply:
Moreover, I think `this.props.resources` could be null at some point.

Reply:
it's required property, up to developer

Reply:
`props.resources` isn't provided by the developer; it's provided by `stripes-connect` when it processes the manifest when this component has `connect` called on it. Using @aliaksei-chumakou's suggestion of `_.get(this.props.resources, 'values.records', [])` or `this.props.resources.values || {}).records || []` are both common patterns.

Reply:
@zburke but developer writes code and it's his responsibility to control all props are passed in right way: `resources` is required property so developer should do everything required from him to pass it to the component

suggestion won't work because of https://github.com/folio-org/stripes-smart-components/blob/657ea72111bf5bc3562fb055d6ed62e3fe222141/lib/ControlledVocab/ControlledVocab.js#L359 I commented previously

Reply:
Your comment that a "developer writes code and its his responsibility to control all props" is technically correct but suggests a misunderstanding of how this component was intended to be used. It would be like describing a sword as perfect for slicing a block of cheese; it's not technically wrong, but we don't typically use swords in the kitchen! Sure, in principle, `resources` could be manually provided by a developer who passes it to an instance of this component. In practice, this prop is never manually passed; [it is provided by stripes-connect](https://github.com/folio-org/stripes-connect/blob/master/doc/api.md#using-the-connected-component) when the component is connected. 

All that said, you're correct that there is already a check in place above at line 359 so the check here is safe. Because that code was already in place it was not part of the diff output and therefore easy to miss. Sorry for dragging this out when your code was technically correct to start with. This code does not follow the common idiom of doing all the validation of `...resources.values.records` in a single line and that threw me off. Please consider this idiom in future PRs.